### ISSUE TITLE
[BUG] - Unable to do model selection on a decreasing metric

### ISSUE THREAD

**Issue by galeone** [zurutech/ashpy#28]
**Describe the bug**
I expect to be able to do model selecting, using the classifier loss as metric. Actually, I find only a JSON with this content:

> cat ~/log/on/best/loss/loss.json 

```
{
    "loss": "-inf",
    "step": "0"
}
```

**Expected behavior**

I should find in the logdir, the best folder containing the JSON + the checkpoint files of the best model, wrt the chosen metric.

**Code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate the problem.

```

def get_model():
    """Create a new autoencoder tf.keras.Model."""
    autoencoder = Autoencoder(
        (64, 64),
        (4, 4),
        kernel_size=3,
        initial_filters=16,
        filters_cap=64,
        encoding_dimension=50,
        channels=3,
    )

    # encoding, representation = autoencoder(input)
    inputs = tf.keras.layers.Input(shape=(64, 64, 3))
    _, reconstruction = autoencoder(inputs)
    model = tf.keras.Model(inputs=inputs, outputs=reconstruction)
    return model


def _train(dataset: tf.data.Dataset, logdir: Path):
    reconstruction_error = ClassifierLoss(tf.keras.losses.MeanSquaredError())
    autoencoder = get_model()

    ClassifierTrainer(
        model=autoencoder,
        optimizer=tf.optimizers.Adam(1e-4),
        loss=reconstruction_error,
        metrics=[ashpy.metrics.ClassifierLoss(model_selection_operator=operator.lt)],
        logdir=str(logdir),
        epochs=100,
    )(dataset, dataset)

```

In short, the initialization of the JSON is wrong when I want to select a model whose metric is decreasing.
